src/block_disagg/block_disagg_read.c

```diff
@@ -36,63 +37,64 @@ __wt_block_disagg_corrupt(
 }
 
 /*
- * __block_disagg_read --
- *     Read an addr/size pair referenced block into a buffer.
+ * __block_disagg_read_multiple --
+ *     Read a page referenced by a page id, checkpoint id, reconciliation id triple into multiple
+ *     buffers.
  */
 static int
-__block_disagg_read(WT_SESSION_IMPL *session, WT_BLOCK_DISAGG *block_disagg, WT_ITEM *buf,
-  WT_PAGE_BLOCK_META *block_meta, uint64_t disagg_id, uint32_t size, uint32_t checksum)
+__block_disagg_read_multiple(WT_SESSION_IMPL *session, WT_BLOCK_DISAGG *block_disagg,
+  WT_PAGE_BLOCK_META *block_meta, uint64_t page_id, uint64_t checkpoint_id,
+  uint64_t reconciliation_id, uint32_t size, uint32_t checksum, WT_ITEM *results_array,
+  u_int *results_count)
 {
     WT_BLOCK_DISAGG_HEADER *blk, swap;
-    size_t bufsize;
+    WT_DECL_RET;
+    WT_ITEM *current;
+
+    WT_UNUSED(reconciliation_id);
 
     if (block_meta != NULL)
         WT_CLEAR(*block_meta);
 
     __wt_verbose(session, WT_VERB_READ, "off %" PRIuMAX ", size %" PRIu32 ", checksum %" PRIu32,
-      (uintmax_t)disagg_id, size, checksum);
+      (uintmax_t)page_id, size, checksum);
 
     WT_STAT_CONN_INCR(session, disagg_block_get);
     WT_STAT_CONN_INCR(session, block_read);
     WT_STAT_CONN_INCRV(session, block_byte_read, size);
 
     /*
-     * Grow the buffer as necessary and read the block. Buffers should be aligned for reading, but
-     * there are lots of buffers (for example, file cursors have two buffers each, key and value),
-     * and it's difficult to be sure we've found all of them. If the buffer isn't aligned, it's an
-     * easy fix: set the flag and guarantee we reallocate it. (Most of the time on reads, the buffer
-     * memory has not yet been allocated, so we're not adding any additional processing time.)
+     * Output buffers do not need to be preallocated, the PALI interface does that.
+     */
+    WT_ERR(block_disagg->plhandle->plh_get(block_disagg->plhandle, &session->iface, page_id,
+      checkpoint_id, results_array, results_count));
+
+    /*
+     * XXX We'll need to handle deltas here.
      */
-    if (F_ISSET(buf, WT_ITEM_ALIGNED))
-        bufsize = size;
-    else {
-        F_SET(buf, WT_ITEM_ALIGNED);
-        bufsize = WT_MAX(size, buf->memsize + 10);
-    }
-    WT_RET(__wt_buf_init(session, buf, bufsize));
-    WT_RET(
-      block_disagg->plhandle->plh_get(block_disagg->plhandle, &session->iface, disagg_id, 0, buf));
+    WT_ASSERT(session, *results_count == 1);
 
+    current = &results_array[0];
     if (block_meta != NULL)
```

Comment:
This should also set the additional fields in the `block_meta`.

Reply:
Fixed, I'm now setting the ones we currently have.